### PR TITLE
Added rich tooltips for 7tv emote links

### PR DESCRIPTION
**Pull request by zneix**
Pull request checklist:

- [x] `CHANGELOG.md` was updated, if applicable

# Description

I've added a new custom resolver for [7tv emotes](https://7tv.app/emotes). It uses their GQL API.
Things left to do:
 * Add proper handling for visibility value by shifting bits from it.
 * Add similar functionality from imgur resolver to show a red text explaining to user that an emote might be not safe for stream and hide the thumbnail.

### CODE REVIEW

File: pkg/resolver/request.go [hunk 1]
```diff
@@ -43,6 +44,18 @@ func RequestGETWithHeaders(url string, extraHeaders map[string]string) (response
...
+func RequestPOST(url, body string) (response *http.Response, err error) {
+	req, err := http.NewRequest("POST", url, strings.NewReader(body))
+	if err != nil {
+		return nil, err
+	}
+
+	req.Header.Set("Content-Type", "application/json")
+
+	return httpClient.Do(req)
+
```
Comment (leon-richardt) [inline pkg/resolver/request.go:56]: extra line detected
```suggestion
```

File: pkg/resolver/request.go [hunk 2]
```diff
@@ -43,6 +44,18 @@ func RequestGETWithHeaders(url string, extraHeaders map[string]string) (response
 	return httpClient.Do(req)
 }
 
+func RequestPOST(url, body string) (response *http.Response, err error) {
+	req, err := http.NewRequest("POST", url, strings.NewReader(body))
```
Comment (leon-richardt) [inline pkg/resolver/request.go:48]: I was reading the [documentation](https://golang.org/pkg/net/http/#NewRequestWithContext) but I'm still not 100% sure. The documentation states:

> If the provided body is also an io.Closer, the returned Request.Body is set to body and will be closed by the Client methods Do, Post, and PostForm, and Transport.RoundTrip.

According to the [`strings.Reader` documentation](https://golang.org/pkg/strings/#Reader), it does not implement the `io.Closer` interface. So do we manually have to close the body?

File: internal/resolvers/seventv/resolver.go [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+	tooltipTemplate = `<div style="text-align: left;">` +
+		`<b>{{.Code}}</b><br>` +
+		`<b>{{.Type}} SevenTV Emote</b><br>` +
+		`<b>By:</b> {{.Uploader}}` +
+		`{{ if .Unlisted }}<li><b><span style="color: red;">UNLISTED</span></b></li>{{ end }}` +
+		`</div>`
+)
+
+var (
+	seventvAPIURL = "https://api.7tv.app/v2/gql"
```
[the remark this answers is not in the document]
Comment (pajlada) [reply internal/resolvers/seventv/resolver.go:26]: Not if we want to add testing :eyes:

[the remark this answers is not in the document]
Comment (pajlada) [reply internal/resolvers/seventv/resolver.go:27]: tests added

File: pkg/resolver/request.go (hunk 2 above)
Comment (pajlada) [reply pkg/resolver/request.go:48]: If it does not implement the `io.Closer` interface it means it does not need to be closed

**Review by pajlada — CHANGES_REQUESTED** [review]
Overall looks good to me - I would want to proxy the thumbnail requests though. Once that's done, I'm happy to have this merged in. Should use the `utils.FormatThumbnailURL` on the current thumbnail format and it should be good, just have to figure out at a way to pass BaseURL to this resolver.

Please take a look at the commits I pushed to the PR too, I added some tests, changed the way the query is generated (using proper GQL variables now), standardized the 'check' function to use the same methods that FrankerFaceZ uses.

**Review by pajlada — APPROVED** [review]
Looks to be working as expected :ok_hand: Feel free to merge this into master whenever you're ready @zneix

**Review by zneix — COMMENTED** [review]
LGTM
ibana.pjg

**Pull request merged**